overlays/tools.nix

```diff
@@ -57,50 +57,31 @@ in { haskell-nix = prev.haskell-nix // {
     cabal-install = "cabal";
   };
 
-  hackage-tool = { name, ... }@args':
+  hackage-tool = projectModule:
     let
-      args = { caller = "hackage-tool"; } // args';
-    in
-      (final.haskell-nix.hackage-package
-        ( # Disable benchmarks and tests by default (since we only want the exe component)
-          { configureArgs = "--disable-benchmarks --disable-tests"; }
-          // args
-          // { name = final.haskell-nix.toolPackageName.${name} or name; }))
-          .getComponent "exe:${final.haskell-nix.packageToolName.${name} or name}";
+      package = final.haskell-nix.hackage-package (
+        (if builtins.isList projectModule then projectModule else [projectModule])
```

Comment:
I have changed `hackage-tool` so it only takes a single module.  If there is a way to compose nix modules we could change the others too (`hackage-package`, `hackage-project`, `cabalProject`, etc.).  Alternatively we could also split out variants that take a list and have wrappers for those that take a single module.  Let me know if you think that would be nicer.

Reply:
I think we suffer a lot from not having types, so I guess the heuristic I end up going for a lot is that an argument named `X` should always have the same type. So I would just use a list of modules everywhere just so it's super consistent, even if that's a bit more annoying in places where you probably only want one?

Reply:
I think it might make sense to move `cabalProject` and `stackProject` code from `overlays/haskell.nix` to `modules/cabal-project.nix` and `overlays/stack-project.nix`.  I think then we might be able to simplify the way we eval the project modules.   However I started trying that and it seems like a much bigger change.  So for now I have changed `hackage-tool` to accept only a list.